routers/web/repo/view.go

```diff
@@ -98,28 +92,12 @@ func findReadmeFileInEntries(ctx *context.Context, entries []*git.TreeEntry) (*n
 		}
 		if i, ok := util.IsReadmeFileExtension(entry.Name(), exts...); ok {
 			log.Debug("Potential readme file: %s", entry.Name())
-			if readmeFiles[i] == nil || base.NaturalSortLess(readmeFiles[i].name, entry.Blob().Name()) {
-				name := entry.Name()
-				isSymlink := entry.IsLink()
-				target := entry
-				if isSymlink {
-					var err error
-					target, err = entry.FollowLinks()
-					if err != nil && !git.IsErrBadLink(err) {
-						return nil, err
-					}
-				}
-				if target != nil && (target.IsExecutable() || target.IsRegular()) {
-					readmeFiles[i] = &namedBlob{
-						name,
-						isSymlink,
-						target.Blob(),
-					}
-				}
+			if readmeFiles[i] == nil || base.NaturalSortLess(readmeFiles[i].Name(), entry.Blob().Name()) {
+				readmeFiles[i] = entry
```

Comment:
I think we still need to check if the readme file is valid here. Assume we have both a broken link readme file and a valid readme file in different directories or names. Moving the logic to `renderReadmeFile` will result in no README file will be displayed.

Reply:
Okay, you're right. I'm going to table this PR and write tests first (like I did for https://github.com/go-gitea/gitea/pull/22196) and then we'll be able to know for sure if this is behaving.

Reply:
Tests are in https://github.com/go-gitea/gitea/pull/23185; the test for this specific case is at 

https://github.com/go-gitea/gitea/blob/8afde8142010985c6d3b6b98eb0be96d7b24da52/tests/integration/repo_test.go#L335-L336

I'll rebase and see if my code passes.

Reply:
Passing in https://drone.gitea.io/go-gitea/gitea/69318/3/10#L848 !